examples/addsvc/client/grpc/client.go

```diff
@@ -0,0 +1,75 @@
+// Package grpc provides a gRPC client for the add service.
+package grpc
+
+import (
+	"time"
+
+	jujuratelimit "github.com/juju/ratelimit"
+	stdopentracing "github.com/opentracing/opentracing-go"
+	"github.com/sony/gobreaker"
+	"google.golang.org/grpc"
+
+	"github.com/go-kit/kit/circuitbreaker"
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/examples/addsvc"
+	"github.com/go-kit/kit/examples/addsvc/pb"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/ratelimit"
+	"github.com/go-kit/kit/tracing/opentracing"
+	grpctransport "github.com/go-kit/kit/transport/grpc"
+)
+
+// New returns an AddService backed by a gRPC client connection. It is the
+// responsibility of the caller to dial, and later close, the connection.
+func New(conn *grpc.ClientConn, tracer stdopentracing.Tracer, logger log.Logger) addsvc.Service {
+	// We construct a single ratelimiter middleware, to limit the total outgoing
+	// QPS from this client to all methods on the remote instance. We also
+	// construct per-endpoint circuitbreaker middlewares to demonstrate how
+	// that's done, although they could easily be combined into a single breaker
+	// for the entire remote instance, too.
+
+	limiter := ratelimit.NewTokenBucketLimiter(jujuratelimit.NewBucketWithRate(100, 100))
+
+	var sumEndpoint endpoint.Endpoint
+	{
+		sumEndpoint = grpctransport.NewClient(
+			conn,
+			"Add",
+			"Sum",
+			addsvc.EncodeGRPCSumRequest,
+			addsvc.DecodeGRPCSumResponse,
+			pb.SumReply{},
+			grpctransport.SetClientBefore(opentracing.FromGRPCRequest(tracer, "Sum", logger)),
+		).Endpoint()
+		sumEndpoint = opentracing.TraceClient(tracer, "Sum")(sumEndpoint)
+		sumEndpoint = limiter(sumEndpoint)
+		sumEndpoint = circuitbreaker.Gobreaker(gobreaker.NewCircuitBreaker(gobreaker.Settings{
+			Name:    "Sum",
+			Timeout: 30 * time.Second,
+		}))(sumEndpoint)
```

Comment:
I also find using multiple expressions more intuitive. I argued for simplifying our APIs in that direction way back in ancient history (https://github.com/go-kit/kit/pull/14#issuecomment-86338127). I wonder if the middleware abstraction has really proved worth it if the only real use of it (`endpoint.Chain`) is avoided because it is less intuitive.

*For reference: `endpoint.Chain` was designed in the discussion on #95 and implemented in it's current form in #96.*